roles/pulp_webserver/tasks/main.yml

```diff
@@ -13,12 +13,23 @@
   tags:
     - always
 
+- name: Properly check that pulp_webserver_tls_key is defined
+  set_fact:
```

Comment:
Please avoid the use of `set_fact`.

Reply:
And the moment we have gone all-in on the ansible collection concept, we can add a test plugin to check for anything that looks empty.
thinking: `when: pulp_webserver_tls_key is not empty`

Reply:
Thanks for the comment, will update to reflect your comment. Makes total sense.

Reply:
Instead of using default in vars, put them in defaults right away. %)

Reply:
Have you tried "empty"? It seems to exist in jinja2. Hence no extra docs @ ansible.

https://www.shellhacks.com/jinja2-check-if-variable-empty-exists-defined-true/

Reply:
@Spredzy ^^

I was just using empty in a playbook by accident, and it is provided in Ansible by Jinja2. I think it make the stuff so much more readable.

Reply:
@mdellweg I am not sure I am entirely following here. Looking at the URL you sent, they are using exactly the same pattern that is being used here.

```
{% if variable is defined and variable|length %}
    variable is defined and not empty
{% else %}
    variable is not defined or empty
{% endif %}
```

> I was just using empty in a playbook by accident, and it is provided in Ansible by Jinja2. I think it make the stuff so much more readable.

Again not fully sure what is meant here. Do you refer to `pulp_webserver_ssl_key: ''` vs. `pulp_webserver_ssl_cert:`.

If so I think we will fall back to the issue this PR is really trying to solve. Idea here, is how - to wrap `pulp/pulp_installer` in a generic way that will understand that my `ssl_cert` or `ssl_key` are actually empty ? 

```
- name: Install Automation Hub via Pulp Installer
  include_role:
    name: "{{ item  }}"
  vars:
    pulp_webserver_tls_cert: "{{ mycertundef | default('') }}"
    pulp_webserver_tls_key: "{{ mykeyundef | default('') }}"
  with_items:
    - "pulp.pulp_installer.pulp_redis"
    - "pulp.pulp_installer.pulp_services"
    - "pulp.pulp_installer.pulp_webserver"
```

I'll happily update the PR - I am just unsure of what (based on the comment) is needed to change

Reply:
Sorry, I must have completely confused myself. I am taking it back.

Reply:
So my earlier review keeps effective.